### ISSUE TITLE
Allow outside audio while in-player mute is active

### ISSUE THREAD

**Issue by J-Stutzmann** [TeamNewPipe/NewPipe#9537]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

The player should not take audio focus while the in-player mute button is active.

To do this, the player should:

- Not Request audio focus on play, when player is muted
- Abandon audio focus on mute, when player is playing
- Request audio focus on unmute, when player is playing
- Not attempt to Request or Abandon audio focus when already in said state

In short, the player should Request audio focus only when moving into the "Playing + Unmuted" state, and Abandon it when moving to any other state.

### Why do you want this feature?

In order to play audio from other apps while NewPipe is muted.

### Additional information

N/A

**Comment by J-Stutzmann** [comment]
In addition, anyone looking to implement this should be aware that Play and playWhenReady have separate audio focus requests.
